models/ipfs_networks.go

```diff
@@ -124,3 +124,12 @@ func (im *IPFSNetworkManager) CreateHostedPrivateNetwork(name, swarmKey string,
 	}
 	return pnet, nil
 }
+
+// Delete is used to remove a network from the database
+func (im *IPFSNetworkManager) Delete(name string) error {
+	net, err := im.GetNetworkByName(name)
+	if err != nil {
+		return err
+	}
+	return im.DB.Unscoped().Delete(net).Error
```

Comment:
why the use of `Unscoped` here?

Reply:
Unless we use `Unscoped` gorm doesn't actually delete the entry from the database, and simply sets the `DeletedAt` field. I believe all this does is prevent gorm from searching for the object, however if you attempt to create a new entry in the database with the same name, we will get an error as the name which is unique has been taken.

See the `Soft Delete` section of the gorm docs http://doc.gorm.io/crud.html#delete